src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -76,26 +77,47 @@ public class NetworkChange
                     {
                         CreateSocket();
                     }
-                    if (s_availabilityTimer == null)
+
+                    if (s_availabilityTimer == null && value != null)
```

Comment:
If we now never add a null value, can we do the value == null check at the beginning of the accessor and not do any of this if it's null?